src/lib/lore.sh

```diff
@@ -352,29 +442,102 @@ function processing_new_patches()
+  for i in $(seq "$starting_index" "$((PATCHSETS_PROCESSED - 1))"); do
     list_of_mailinglist_patches["$i"]=$(< "${shared_dir_for_parallelism}/${i}")
   done
 }
 
+# This function is primarily a mediator to manage the complex action of fetching
+# the lastest patchsets from a public mailing list archived on lore.kernel.org.
+# The fetching of patchsets has 3 steps:
+#  1. Build a lore query URL to match only messages that are patches from a given
+#     timeframe.
+#  2. Make a request to the URL built in step to obtain a list of patches from
+#     the respective timeframe ordered by the recieved time of the patch on the
+#     lore.kernel.org servers.
+#  3. Process the list of patches to a list of patchsets stored in the
+#     `list_of_mailinglist_patches` array.
+#
+# In case the number of patchsets in `list_of_mailinglist_patches` is less than
+# `NUMBER_OF_PATCHSETS_PER_FETCH`, get a adjacent timeframe chunk and repeat steps
+# 1 to 3.
+#
+# Each entry in `list_of_mailinglist_patches` has the following patchset metadata
+# separated by `SEPARATOR_CHAR`:
+#   author name, author email, patchset version, number of patches, patch title, message-ID
+#
+# @target_mailing_list: A string name that matches the mailing list name
+#   registered to lore
+# @flag: Flag to control function output
+#
+# Return:
+# If either step 1 or 2 fails, returns the error code from these steps, and 0, otherwise.
+function fetch_latest_patchsets_from()
+{
+  local target_mailing_list="$1"
+  local flag="$2"
+  local raw_xml
+  local lore_query_url
+  local days
+  local last_patch_timestamp
+  local xml_result_file_name
+  local pre_processed_patches
+  local ret
+
+  # Assigning starting values.
+  days="$TIMEFRAME_IN_DAYS"
+  last_patch_timestamp=''
+  xml_result_file_name="${target_mailing_list}-patches.xml"
+  PATCHSETS_PROCESSED=0
+  flag=${flag:-'SILENT'}
+
+  while [[ "$PATCHSETS_PROCESSED" -lt "$NUMBER_OF_PATCHSETS_PER_FETCH" ]]; do
+    # Building URL for querying lore servers for a xml file with patches.
+    lore_query_url=$(compose_lore_query_url_with_verification "$target_mailing_list" "$days" "$last_patch_timestamp")
+    ret="$?"
+    [[ "$ret" != 0 ]] && return "$ret"
+
+    # Request xml file with patches.
+    download "$lore_query_url" "$xml_result_file_name" "$CACHE_LORE_DIR" "$flag"
+    ret="$?"
+    [[ "$ret" != 0 ]] && return "$ret"
+
+    # If the resulting file doesn't contain any patches, it will be an html file.
+    # In this case, we expand the timeframe and make another fetch.
+    if is_html_file "${CACHE_LORE_DIR}/${xml_result_file_name}"; then
+      days="$((days + TIMEFRAME_IN_DAYS))"
+      continue
+    fi
+
+    raw_xml=$(< "${CACHE_LORE_DIR}/${xml_result_file_name}")
+
+    # Processing patches into patchsets that will be stored in `list_of_mailinglist_patches`.
+    pre_processed_patches=$(pre_process_xml_result "$raw_xml")
+    # TODO: Is passing `pre_processed_patches` (huge string) as argument a possible bottleneck?
```

Comment:
What do you consider huge?

Reply:
Don't know if this is huge, but in the ten thousands lines and one hundred thousands words mark.